### ISSUE TITLE
charts.d: retry failed modules, if they had collected data in the past

### ISSUE THREAD

**Issue by centminmod** [firehol/netdata#239]
Curious how do i troubleshoot this error ? and what is being disabled specifically ? As i still see nginx live updates in the dashboard ?
```
tail -2 /var/log/netdata/error.log   
16-04-15 15:09:32: INFO: netdata: executing command: 'exec /usr/libexec/netdata/plugins.d/tc-qos-helper.sh 1' on pid 24184.
charts.d: chart 'nginx' update() function reports failure. Disabling it.
```

cheers

George

**Comment by ktsaou** [comment]
The plugin failed to collect data from nginx and disabled itself. Possibly you restarted nginx.
When this happens, the nginx charts on the dashboard should not be adding more data.

I have a todo to fix that: When an module that was able to collect data, encounters an error, it should retry a few times before giving up. Now, the failed modules never re-try. As a quick fix, I have instructed all external plugins to restart every 4 hours. So, even if something fails, after at most 4 hours, it will retry.

**Comment by centminmod** [comment]
Ah yes i did restart nginx server :)

Definitely need a fix for this. Cheers @ktsaou

**Comment by ktsaou** [comment]
Added a fix with #525 
Thanks!
